django/conf/__init__.py

```diff
@@ -183,6 +186,11 @@ def __init__(self, settings_module):
                     raise ImproperlyConfigured(
                         "The %s setting must be a list or a tuple." % setting
                     )
+
+                if setting in int_settings and not isinstance(setting_value, int):
+                    raise ImproperlyConfigured(
+                        "The %s setting must be an integer." % setting
+                    )
```

Comment:
We have dozens of settings and we cannot add type checks for all of them to the `Settings`. Especially when an expected type is clearly documented.

Reply:
Fair enough.